rwslib/builders/modm.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'glow'
+
+import datetime
+import enum
+
+
+class MODMExtensionRegistry(enum.Enum):
+    """
+    A registry of MODM extension Elements
+    TODO: Get this from the Schema
+    """
+    StudyEventDef = ["ArmAssociation"]
+    StudyEventRef = ["ArmAssociation"]
+    ClinicalData = ["ExternalStudyID", "StudyUUID", "AuditSubCategoryName",
+                    "StudyName", "ClientDivisionUUID", "ClientDivisionSchemeUUID",
+                    "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                    "IsSDVRequired", "IsSDVComplete"]
+    StudyEventData = ["StartWindowDate", "EndWindowDate", "StudyEventUUID",
+                      "InstanceName", "VisitTargetDate", "InstanceId",
+                      "InstanceOverDue", "InstanceStartWindow", "InstanceEndWindow",
+                      "InstanceClose", "InstanceAccess", "StudyEventDate",
+                      "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                      "VisitFirstDataEntryDate",
+                      "IsSDVRequired", "IsSDVComplete"]
+    SubjectData = ["SubjectName", "Status",
+                   "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                   "IsSDVRequired", "IsSDVComplete"]
+    FormData = ["FormUUID", "DataPageName", "DataPageID",
+                "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                "IsSDVRequired", "IsSDVComplete"]
+    ItemGroupData = ["ItemGroupUUID", "RecordID",
+                     "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                     "IsSDVRequired", "IsSDVComplete"]
+    ItemData = ["ItemUUID",
+                "SDRCompleteDate", "SDVCompleteDate", "LockCompleteDate",
+                "IsSDVRequired", "IsSDVComplete"]
+    SiteRef = ["SiteStartDate", "SiteCloseDate", "LocationOIDType"]
+    Location = ["SiteStartDate", "SiteCloseDate"]
+
+
+class MODMAttribute(object):
+    def __init__(self, attribute, value):
+        self.attribute = attribute
+        self.raw_value = value
+
+    @property
+    def tag(self):
+        return "mdsol:{}".format(self.attribute)
+
+    @property
+    def value(self):
+        if isinstance(self.raw_value, (datetime.datetime, datetime.date)):
+            return self.raw_value.isoformat()
+        return self.raw_value
```

Comment:
Can it be a boolean value you'd need to convert to string?

Reply:
So, a boolean would map to the YesNoRave codelist?

Reply:
I don't know. Is there a rule for MODM attributes? What is used for bools? TRUE/FALSE ?

Reply:
Ultimately it's a little Rave oriented, I think the Yes/No should probably work.  Will add.

Reply:
There's no schema for the MODM attributes?